md-dao/src/main/java/com/wipro/ats/bdre/md/dao/JobDAO.java

```diff
@@ -376,14 +371,14 @@
                 for (Object subProcessObject : listOfSubProcessCriteria.list()) {
                     Process subProcess = (Process) subProcessObject;
                     Criteria listOfTargetBatchIdCriteria = session.createCriteria(BatchConsumpQueue.class)
-                            .add(Restrictions.eq("process", subProcess));
-                    if (listOfTargetBatchIdCriteria.list().size() > 0) {
+                            .add(Restrictions.eq(PROCESS, subProcess));
+                    if (!listOfTargetBatchIdCriteria.list().isEmpty()) {
                         BatchConsumpQueue targetBatchOfBCQ = (BatchConsumpQueue) listOfTargetBatchIdCriteria.list().get(0);
                         listOfTargetBatchId.add(targetBatchOfBCQ.getBatchByTargetBatchId().getBatchId());
                     }
                 }
 
-                if (listOfTargetBatchId.size() != 0) {
+                if (!listOfTargetBatchId.isEmpty()) {
                     Criteria batchIdCriteria = session.createCriteria(Batch.class).add(Restrictions.in("batchId", listOfTargetBatchId));
                     for (Object batchIdObject : batchIdCriteria.list()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)